[1749] Add Allocations documentation (#1657)
DFE-Digital/publish-teacher-training: docs/allocations.md

# Allocations

> Each year, around the same time as [Rollover](./rollover.md), providers request

Reviewer: ```suggestion
Each academic year, around the same time as [Rollover](./rollover.md), providers request
```

I think we need make a slight distinction with what is that `year`.

places for courses that have numbers of applicants limited as they are generally
over-supplied. This process is called 'Allocations'.

This document lists the changes needed to be made to the Publish codebase and
the timings for these changes.

## Before Allocations

> - Update any hardcoded dates relating to Allocation years and timelines

Reviewer: I think we need to make some references to academic year

Author: done!

# Allocations

> Each year, around the same time as [Rollover](./rollover.md), providers request
> places for courses that have numbers of applicants limited as they are generally
> over-supplied. This process is called 'Allocations'.

Reviewer: ```suggestion
Each year, around the same time as [Rollover](./rollover.md), providers request
places for courses. This is to limited applicants as they are generally
over-supplied for that particular type of courses. This process is called 'Allocations'.
```

This document lists the changes needed to be made to the Publish codebase and
the timings for these changes.

## Before Allocations

- Update any hardcoded dates relating to the academic year the Allocations are
  for e.g. Allocations made in the year 2021 will be for the academic year

>   2022 to 2023. [TBC - this may be refactored to be dynamic based on a setting]

Reviewer: @mooktakim :eyes:

Author: I've updated!
